internal/v2/application/command.go

```diff
@@ -373,6 +381,12 @@ func (c *CommandProcessor) WriteCommand() edgexErr.EdgeX {
 
 		reqs[i].DeviceResourceName = cv.DeviceResourceName
 		reqs[i].Attributes = dr.Attributes
+		if c.attributes != "" {
+			if len(reqs[i].Attributes) <= 0 {
+				reqs[i].Attributes = make(map[string]string)
+			}
+			reqs[i].Attributes[common.URLRawQuery] = c.attributes
+		}
```

Comment:
This code snippet are located in three places of this file.  Please extract it to another func.  You can call it `extractAttributes`